simapp/simd/cmd/root.go

```diff
@@ -192,6 +197,7 @@ func newApp(logger log.Logger, db dbm.DB, traceStore io.Writer, appOpts serverty
 		cast.ToString(appOpts.Get(flags.FlagHome)),
 		cast.ToUint(appOpts.Get(server.FlagInvCheckPeriod)),
 		simapp.MakeTestEncodingConfig(), // Ideally, we would reuse the one created by NewRootCmd.
+		appOpts,
```

Comment:
Can we remove `cast.ToString(appOpts.Get(flags.FlagHome))`, `cast.ToUint(appOpts.Get(server.FlagInvCheckPeriod))`, and use `appOpts` directly to read the values inside `NewSimApp`? Let's minimize the number of arguments when possible. May be not relavant for this PR.

Reply:
Yeah, I was planning to group all this config options already long time ago. 
Should I do it in this PR or in a new one? Thoughts?

Reply:
A separate PR works too, can you create an issue to track this?

Reply:
Yes, I  think we should do it in a separate post, because this will be another breaking feature. Also, not sure if that will land in release,  @clevinson, @aaronc  , what do you think?

Reply:
Here is the old issue I created it. I'm happy that we like to see a reduction of this arguments. It's an easy change.
https://github.com/cosmos/cosmos-sdk/issues/7178

Reply:
Well we should still keep the variadic options (which is what I was trying to advocate for in #7178 -- I'm against options structs for ALL arguments in this case). However, for the sake of this PR, we can at least avoid the following calls/args:

```
cast.ToString(appOpts.Get(flags.FlagHome)),
cast.ToUint(appOpts.Get(server.FlagInvCheckPeriod)),
```

Just pass `appOpts` and let the app do the above two calls. This is what I meant.